src/lib/components/Text/Text.js

```diff
@@ -41,6 +41,10 @@ const wordToSpan = (word, config, active, toggleActive) => {
     classes.push(styles.active);
   }
 
+  if (highlighted.has(id)) {
```

Comment:
This is where highlighting is added to highlighted words. For every word in the sentence, the `highlighted` Set is checked for that word's ID. If it is present, the `highlighted` class is added.

The reason that this is a Set instead of an array, is that the `Set.has` function is performed in constant time, whereas the `Array.includes` function takes linear time. In other words, `Set.has` is done in a single computation, whereas `Array.includes` performs `Array.length` computations.

So, hypothetically, if there were a sentence of 200 words and 100 of them were highlighted, this code would take roughly 200 computations (it runs once for every word in the sentence; 200 X 1 = 200). If it were an array, then for every word, the `Array.includes` function would take an average of 100 computations. This would result in 100 X 200 = 20,000 computational steps, two orders of magnitude greater than the code using a set.

Reply:
Thank you for a such detailed exaple here - I also found an article that confirms your calculations about include inside another array
https://dev.to/arnaud/using-array-prototype-includes-vs-set-prototype-has-to-filter-arrays-41fg

Would think about my code